packages/react-app/src/CreateInk.js

```diff
@@ -320,370 +322,417 @@ useEffect(() => {
 
     }
 
-};
+  };
 
 
-const onFinishFailed = errorInfo => {
-  console.log('Failed:', errorInfo);
-};
+  const onFinishFailed = errorInfo => {
+    console.log('Failed:', errorInfo);
+  };
 
-const triggerOnChange = (lines) => {
-  let saved = JSON.stringify({
-    lines: lines,
-    width: drawingCanvas.current.props.canvasWidth,
-    height: drawingCanvas.current.props.canvasHeight
-  });
+  const triggerOnChange = (lines) => {
+    let saved = JSON.stringify({
+      lines: lines,
+      width: drawingCanvas.current.props.canvasWidth,
+      height: drawingCanvas.current.props.canvasHeight
+    });
+
+    drawingCanvas.current.loadSaveData(saved, true);
+    //setLoadedLines(lines.length)
+    //setInitialDrawing(saved)
+    drawingCanvas.current.lines = lines;
+    saveDrawing(drawingCanvas.current, true);
+  };
 
-  drawingCanvas.current.loadSaveData(saved, true);
-  //setLoadedLines(lines.length)
-  //setInitialDrawing(saved)
-  drawingCanvas.current.lines = lines;
-  saveDrawing(drawingCanvas.current, true);
-};
+  const undo = () => {
+    if (!drawingCanvas.current.lines.length) return;
 
-const undo = () => {
-  if (!drawingCanvas.current.lines.length) return;
+    if (drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].ref) {
+      drawingCanvas.current.lines[0].brushColor = drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].brushColor;
+      let lines = drawingCanvas.current.lines.slice(0, -1);
+      triggerOnChange(lines);
+    } else {
+      let lines = drawingCanvas.current.lines.slice(0, -1);
+      triggerOnChange(lines);
+    }
+  };
 
-  if (drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].ref) {
-    drawingCanvas.current.lines[0].brushColor = drawingCanvas.current.lines[drawingCanvas.current.lines.length - 1].brushColor;
-    let lines = drawingCanvas.current.lines.slice(0, -1);
-    triggerOnChange(lines);
-  } else {
-    let lines = drawingCanvas.current.lines.slice(0, -1);
+  const downloadCanvas = async () => {
+    const myData = drawingCanvas.current.lines; // I am assuming that "this.state.myData"
+    // is an object and I wrote it to file as
+    // json
+    const fileName = "canvas";
```

Comment:
could we maybe include the timestamp here so users can download different files?